pyproject.toml

```diff
@@ -1,6 +1,7 @@
 [build-system]
-requires = [
-    "setuptools",
-    "cython>=0.25",
-]
-build-backend = "setuptools.build_meta"
\ No newline at end of file
+requires = ["maturin>=0.13,<0.14"]
+build-backend = "maturin"
```

Comment:
Unless there's a particular reason to use maturin, I think that `setuptools-rust` will be easier to use with wheelwright, and I've had problems in the past getting maturin to generate working sdists (but this was a while ago, and partially related to relative source paths, I think).

Reply:
I have gone back and forth a few times, but I don't know why I ended up with maturin, so I'll try to convert it to `setuptools-rust`.

Reply:
We will have to modernize wheelwright a bit at some point, but for now it's still using `python setup.py sdist`.

Reply:
But the main thing is "working sdist" part.

Reply:
The current version doesn't work on my end at least. (And the name `rust_ops` was unexpected?)

```
        Compiling sentencepiece v0.10.0
      error: could not compile `sentencepiece` due to 2 previous errors
      💥 maturin failed
        Caused by: Failed to build a native library through cargo
        Caused by: Cargo build finished with "exit status: 101": `"cargo" "rustc" "--release" "--message-format" "json" "--lib"`
      📦 Including license file "LICENSE"
      🍹 Building a mixed python/rust project
      🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.7
      🐍 Not using a specific python interpreter
      error[E0277]: the trait bound `f32: std::cmp::Eq` is not satisfied
         --> /home/adriane/.cargo/registry/src/github.com-1ecc6299db9ec823/sentencepiece-0.10.0/src/sentencepiece.rs:15:5
          |
      3   | #[derive(Clone, Eq, PartialEq, Message)]
          |                 -- in this derive macro expansion
      ...
      15  |     pub score: Option<f32>,
          |     ^^^^^^^^^^^^^^^^^^^^^^ the trait `std::cmp::Eq` is not implemented for `f32`
          |
          = note: required because of the requirements on the impl of `std::cmp::Eq` for `Option<f32>`
      note: required by a bound in `AssertParamIsEq`
          = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)
      
      
      error: aborting due to previous error
```

(I kind of enjoyed the lack-of-other-complicating-stuff in the python version?)

Reply:
I am actually really not sure we want to do this at all?

Reply:
The thing is that implementing things like BPE (which I am working on now) really gets terrible in Cython, because you need binary heaps, string views, etc. to do it efficiently. So you end up with Cython/C++ template soup.

Reply:
> The current version doesn't work on my end at least.

Still working on things, it's a draft, not really ready yet to try out or review. Just wanted to do some CI runs.

Reply:
All I tried out was building from the sdist. I wanted something that was easier to build than the existing `sentencepiece`, otherwise this seems like it's starting to having fewer advantages over `tokenizers`?

Reply:
That's a fair point. I'll rethink a bit how to do BPE. I don't want to do it in Python, because it will be too slow. But I think in template-heavy Cython is also messy. Maybe I'll do it in pure C++ and call that from Cython.

Reply:
Pure C++ is fine with me. I'm more worried about more rust dependencies. It's just so hard for users from the python side.